examples/spmm/spmm.cc

```diff
@@ -946,6 +1035,39 @@ int main(int argc, char **argv) {
     double Adensity = nan("undefined");
     double Bdensity = nan("undefined");
 
+    std::string PStr(getCmdOption(argv, argv + argc, "-P"));
+    P = parseOption(PStr, P);
+    std::string QStr(getCmdOption(argv, argv + argc, "-Q"));
+    Q = parseOption(QStr, Q);
+
+    if (P * Q != mpi_size) {
+      if (!cmdOptionExists(argv, argv + argc, "-Q") && (mpi_size % P) == 0)
+        Q = mpi_size / P;
+      else if (!cmdOptionExists(argv, argv + argc, "-P") && (mpi_size % Q) == 0)
+        P = mpi_size / Q;
+      else {
+        if (0 == mpi_rank) {
+          std::cerr << P << "x" << Q << " is not a valid process grid -- bailing out" << std::endl;
+          MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
+        }
+      }
+    }
+
+    std::string seedStr(getCmdOption(argv, argv + argc, "-s"));
+    unsigned int seed = parseOption(seedStr, 0);
```

Comment:
There is a problem here: `parseOption` uses `std::stoi` to parse the string but the provided seed is `unsigned int`, potentially leading to an out-of-range exception if passed back in. One solution is to use `std::stol` in `parseOption` :)